sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationClientBuilder.java

```diff
@@ -0,0 +1,386 @@
+     */
+    private ElevationClientImpl buildInnerClient() {
+        if (endpoint == null) {
+            this.endpoint = "https://atlas.microsoft.com";
+        }
+        if (serviceVersion == null) {
+            this.serviceVersion = ElevationServiceVersion.getLatest();
+        }
+        if (pipeline == null) {
+            this.pipeline = createHttpPipeline();
+        }
+        // client impl
+        ElevationClientImplBuilder builder = new ElevationClientImplBuilder();
+        builder.host(this.endpoint);
+        builder.apiVersion(this.serviceVersion.getVersion());
+        builder.pipeline(this.pipeline);
+        builder.clientId(this.elevationClientId);
+        builder.httpClient(this.httpClient);
+        builder.httpLogOptions(this.httpLogOptions);
+
+        ElevationClientImpl client = builder.buildClient();
+        return client;
+    }
+
+    private HttpPipeline createHttpPipeline() {
+        Configuration buildConfiguration =
+            (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        if (httpLogOptions == null) {
+            httpLogOptions = new HttpLogOptions();
+        }
+        if (clientOptions == null) {
+            clientOptions = new ClientOptions();
+        }
+
+        // Configure pipelines and user agent
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+        String clientName = properties.getOrDefault(SDK_NAME, "JavaTrafficSDK");
+        String clientVersion = properties.getOrDefault(SDK_VERSION, serviceVersion.getVersion());
+        String applicationId = CoreUtils.getApplicationId(clientOptions, httpLogOptions);
+        policies.add(new UserAgentPolicy(applicationId, clientName, clientVersion, buildConfiguration));
+
+        // configure headers
+        HttpHeaders headers = new HttpHeaders();
+        clientOptions.getHeaders().forEach(header -> headers.set(header.getName(), header.getValue()));
+        if (headers.getSize() > 0) {
+            policies.add(new AddHeadersPolicy(headers));
+        }
+
+        // Authentications
+        if (tokenCredential != null) {
+            if (this.elevationClientId == null) {
+                throw LOGGER.logExceptionAsError(
+                    new IllegalArgumentException("Missing 'elevationClientId' parameter required for Azure AD Authentication"));
+            }
+            // we need the x-ms-client header
+            HttpHeaders clientHeader = new HttpHeaders();
+            clientHeader.add(X_MS_CLIENT_ID, this.elevationClientId);
+            policies.add(new AddHeadersPolicy(clientHeader));
+
+            // User token based policy
+            policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPES));
+        } else if (keyCredential != null) {
+            policies.add(new AzureKeyCredentialPolicy(MAPS_SUBSCRIPTION_KEY, keyCredential));
+        } else {
+            // Throw exception that credential and tokenCredential cannot be null
```

Comment:
Should also throw if multiple credentials are supplied rather than using one

Reply:
This will be updated in the next version of beta for consistency across modules that have been released